File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int major = nums[0];
        int count = 1;
        
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == major) {
                count++;
            } else {
                count--;
            }
            if (count == 0) {
                major = nums[i];
                count = 1;
            }
        }
        
        return major;
    }
};

File: 0169-majority-element/0169-majority-element.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var ans = nums[0]
        var count =1
        for(i in 0 until nums.size){        
            if(nums[i]==ans){
                count++
            }else{
                count--;
            }
            if(count==0){
                ans = nums[i]
                count=1
            }
        }
        return ans
    }
}